Parser/string_parser.c

```diff
@@ -61,15 +63,15 @@ warn_invalid_escape_sequence(Parser *p, const char *first_invalid_escape, Token
             p->known_err_token = t;
             if (octal) {
                 RAISE_SYNTAX_ERROR(
-                    "\"\\%.3s\" is an invalid octal escape sequence. "
-                    "Did you mean \"\\\\%.3s\"? A raw string is also an option.",
+                    "\"\\%o\" is an invalid octal escape sequence. "
+                    "Did you mean \"\\\\%o\"? A raw string is also an option.",
                     first_invalid_escape, first_invalid_escape);
             }
             else {
                 RAISE_SYNTAX_ERROR(
                     "\"\\%c\" is an invalid escape sequence. "
                     "Did you mean \"\\\\%c\"? A raw string is also an option.",
-                    c, c);
+                    first_invalid_escape, first_invalid_escape);
```

Comment:
double checking: in C, does passing an int when formatting with %c always work?  is a single char parameter expected to come from the platforms lowest significant byte of int sized space per in C calling conventions?  casting these down to `(unsigned char)` might be a good idea for clarity even if it is defined behavior.
Same comment for the other files where this is done.

Reply:
The expected argument type for `%c` in `printf()` is `int`. The expected argument type for `%c` in `PyUnicode
_Format()` (which is used here) is also `int`.